x-pack/plugins/aiops/public/application/utils/query_utils.ts

```diff
@@ -48,10 +50,23 @@ export function buildBaseFilterCriteria(
     filterCriteria.push(query);
   }
 
-  if (selectedChangePoint && includeSelectedChangePoint) {
-    filterCriteria.push({
-      term: { [selectedChangePoint.fieldName]: selectedChangePoint.fieldValue },
-    });
+  const groupFilter = [];
+  if (selectedGroup) {
+    for (const fieldName in selectedGroup.group) {
+      if (selectedGroup.group.hasOwnProperty(fieldName)) {
+        groupFilter.push({ term: { [fieldName]: selectedGroup.group[fieldName] } });
+      }
+    }
+  }
```

Comment:
This needs to consider `repeatedValues` like this:

```
const groupFilter = [];
  if (selectedGroup) {
    const allItems = { ...selectedGroup.group, ...selectedGroup.repeatedValues };
    for (const fieldName in allItems) {
      if (allItems.hasOwnProperty(fieldName)) {
        groupFilter.push({ term: { [fieldName]: allItems[fieldName] } });
      }
    }
  }
  ```

Reply:
Updated in 655ffec956f353b5c6306d305719c1fa8788d48b